### DISCUSSION TITLE
Disable and enable POE-port on specific times

### DISCUSSION THREAD

**Discussion by joppez** [Art-of-WiFi/UniFi-API-client#139] [Q&A]
I have been looking for running a script to disable and enable a POE-port at night and in the morning (during sleeptime).

I found the "update_switch_poe-mode.php" example, but I'm lost as to where I need to start get the script running and how I can set the times.

I have the api browser running on my Raspberry Pi and that works. But any help as where to start now to get the script running would be very helpfull.

**Comment by malle-pietje** [comment]
I suggest starting to get one or more copies of that script running from the command line.
```
php update_switch_poe-mode.php off
``` 
or
```
php update_switch_poe-mode.php on
```
Make sure to configure the correct MAC address for the switch/IW-device as well as the switch ports:
https://github.com/Art-of-WiFi/UniFi-API-client/blob/master/examples/update_switch_poe-mode.php#L32
https://github.com/Art-of-WiFi/UniFi-API-client/blob/master/examples/update_switch_poe-mode.php#L37

Then create a cron job that kicks off your command. This is out of the scope of the support I provide here but maybe others can pitch in with suggestions and or tips.

**Comment by joppez** [comment]
Thanks for the reply. I put the php file into the /var/www/html folder. Next I have updated the MAC address and the number of the port.

When I execute the php file ony my raspberry, it says:

_PHP Warning:  require_once(config.php): failed to open stream: No such file or directory in /var/www/html/update_switch_poe-mode.php on line 11
PHP Fatal error:  require_once(): Failed opening required 'config.php' (include_path='.:/usr/share/php') in /var/www/html/UniFi-API-browser/update_switch_poe-mode.php on line 11_

What I don't understand: where do I need to put the php file and how does the php-file connect to the Unifi Api? I'm sorry if this are really basic questions.

**Reply by joppez** [reply]
Any help would be welcome :)

**Reply by thib3113** [reply]
Ok, so basically : 
 - first maybe create a subfolder insteed of directly using the html folder ( like `/var/www/html/unifi` ) .
 - in the example folder, take the `config.template.php`, fill the values (that will tell how to connect to your unifi), and rename the file config.php
 - in the same folder, put your update_switch_poe-mode.php .
 
 ---
 the line in https://github.com/Art-of-WiFi/UniFi-API-client/blob/master/examples/update_switch_poe-mode.php#L22 tell to require (require and require_once are nearly the same) the file `config.php`, it's a relative path, to 'config.php' (so in the same folder as the other php) . 
 
ALso reading the comment just before the require, it tell you : 
```
 * include the config file (place your credentials etc. there if not already present)
 * see the config.template.php file for an example
```
so : 
 - you need to place your credential in a `config.php` file
 - you can take the file config.template.php for the example of "how to write your credentials"


For the question `how does the php-file connect to the Unifi Api ?`, on line [65](https://github.com/Art-of-WiFi/UniFi-API-client/blob/master/examples/update_switch_poe-mode.php#L65), it will pass 
`$controlleruser`, `$controllerpassword`, `$controllerurl`, `$site_id`, and `$controllerversion`, all of this variables come from the config.php .
